rust/scannerctl/src/interpret/mod.rs

```diff
@@ -171,6 +174,23 @@ where
     Ok(result)
 }
 
+fn create_loader_from_feed_json(
+    store: &DefaultDispatcher,
+    path: PathBuf,
+) -> Result<FSPluginLoader<PathBuf>, CliError> {
+    let result = FSPluginLoader::new(path.clone());
+    let buf = fs::read_to_string(&path).map_err(|e| CliError::load_error(e, &path))?;
+    let vts: Vec<storage::item::Nvt> = serde_json::from_str(&buf)?;
```

Comment:
The approach of loading the file as a string into heap, use that heap to serialize into a struct by using serde_json::from_str and then iterate to map it seems relatively expensive.

A better approach could be to use the file reader inside serde directly and may using non collecting approach to immediately map it into a map.

I'm going to provide an example later, writing it now so that I don't forget it.

Reply:
after testing serde_json::from_str and iterate on that is faster than creating a bufreader and use from_reader.